flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/KnnModel.java

```diff
@@ -0,0 +1,378 @@
+     *
+     * @param env Stream execution environment.
+     * @param path Model path.
+     * @return Knn model.
+     */
+    public static KnnModel load(StreamExecutionEnvironment env, String path) throws IOException {
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+        KnnModel retModel = ReadWriteUtils.loadStageParam(path);
+
+        Source<Row, ?, ?> source =
+                FileSource.forRecordStreamFormat(
+                                new KnnModelData.ModelDataStreamFormat(),
+                                ReadWriteUtils.getDataPaths(path))
+                        .build();
+        DataStream<Row> modelDataStream =
+                env.fromSource(source, WatermarkStrategy.noWatermarks(), "data");
+        retModel.modelData =
+                new Table[] {tEnv.fromDataStream(modelDataStream, KnnModelData.getModelSchema())};
+        return retModel;
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictOperator
+            extends AbstractUdfStreamOperator<Row, AbstractRichFunction>
+            implements OneInputStreamOperator<Row, Row> {
+
+        private boolean firstEle = true;
+        private final String featureCol;
+        private transient KnnModelData modelData;
+        private final Integer topN;
+        private final String broadcastKey;
+
+        public PredictOperator(String broadcastKey, int k, String featureCol) {
+            super(new AbstractRichFunction() {});
+            this.topN = k;
+            this.broadcastKey = broadcastKey;
+            this.featureCol = featureCol;
+        }
+
+        @Override
+        public void processElement(StreamRecord<Row> streamRecord) {
+            Row value = streamRecord.getValue();
+            output.collect(new StreamRecord<>(map(value)));
+        }
+
+        private Row map(Row row) {
+            if (firstEle) {
+                loadModel(userFunction.getRuntimeContext().getBroadcastVariable(broadcastKey));
+                firstEle = false;
+            }
+            DenseVector vector = (DenseVector) row.getField(featureCol);
+            Tuple2<List<Integer>, List<Double>> t2 = findNeighbor(vector, topN, modelData);
+            Row ret = new Row(row.getArity() + 1);
+            for (int i = 0; i < row.getArity(); ++i) {
+                ret.setField(i, row.getField(i));
+            }
+
+            ret.setField(row.getArity(), getResult(t2));
+            return ret;
+        }
+
+        /**
+         * Finds the nearest topN neighbors from whole nodes.
+         *
+         * @param input Input vector.
+         * @param topN Top N.
+         * @return Neighbors.
+         */
+        private Tuple2<List<Integer>, List<Double>> findNeighbor(
+                DenseVector input, Integer topN, KnnModelData modelData) {
+            PriorityQueue<Tuple2<Double, Integer>> priorityQueue =
+                    new PriorityQueue<>(modelData.getQueueComparator());
```

Comment:
Would it be simpler to instantiate `Comparator.comparingDouble(o -> -o.f0)` as a member field of `PredictOperator` directly instead of calling `getQueueComparator(...)`?

Reply:
OK